pkg/cmd/server/admin/decrypt.go

```diff
@@ -0,0 +1,158 @@
+package admin
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+
+	"github.com/openshift/origin/pkg/cmd/util"
+	"github.com/spf13/cobra"
+
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+
+	configapi "github.com/openshift/origin/pkg/cmd/server/api"
+	pemutil "github.com/openshift/origin/pkg/cmd/util/pem"
+)
+
+const DecryptCommandName = "decrypt"
+
+type DecryptOptions struct {
+	Data    *pem.Block
+	InFile  string
+	OutFile string
+
+	Key       []byte
+	KeyArg    string
+	InKeyFile string
+
+	Input  io.Reader
+	Output io.Writer
+	Stderr io.Writer
+}
+
+const decryptExample = `	# Decrypt using a key file, and output the decrypted value:
+	$ %[1]s --in=secret1.data --keyin=secret1.key
+	
+	# Decrypt using a key file, writing the decrypted value to a file:
+	$ %[1]s --in=secret2.data --keyin=secret1.key --out=secret2.txt
+
+	# Decrypt with a prompted password:
+	$ %[1]s --in=secret3.data
+`
+
+func NewCommandDecrypt(commandName string, fullName, encryptFullName string, out io.Writer) *cobra.Command {
+	options := &DecryptOptions{Output: out}
+
+	cmd := &cobra.Command{
+		Use:     commandName,
+		Short:   fmt.Sprintf("Decrypt data encrypted with %q", encryptFullName),
+		Example: fmt.Sprintf(decryptExample, fullName),
+		Run: func(cmd *cobra.Command, args []string) {
+			kcmdutil.CheckErr(options.Validate(args))
```

Comment:
So I can see value in validating args for giving nice messages, but validation on the options struct is about allowing a delegator (who will likely *not* have valid args) know whether his options are likely to work.  If you want both, make both.